chore(deps): bump entities from 4.5.0 to 6.0.0

cardoso opened the pull request:
Closes #1296

cardoso commented:
Tests are passing locally, however I need to take a second look into build:cjs. I don't use commonjs, but not looking to break anything unless needed. It's probably a simple tweak.

43081j commented on packages/parse5-htmlparser2-tree-adapter/package.json:
```diff
@@ -26,7 +26,7 @@
     },
     "scripts": {
-        "build:cjs": "tsc --moduleResolution node10 --module CommonJS --target ES6 --outDir dist/cjs && echo '{\"type\":\"commonjs\"}' > dist/cjs/package.json"
+        "build:cjs": "tsc --moduleResolution node16 --module node16 --target ES6 --outDir dist/cjs && echo '{\"type\":\"commonjs\"}' > dist/cjs/package.json"
```
im not sure you can change `module` here since `node16` will detect that it is `type: "module"` and output ESM

`--module CommonJS` was forcing it to output commonjs regardless of the `type`

cardoso replied:
Thanks for the review 🙏

Node10 module resolution wasn't working after the change, and module CommonJS can't be paired with Node16 resolution.

The other PR ran into the same thing.

I'll find a proper solution tomorrow.

fb55 replied:
I looked into migrating to tshy to resolve, which seemed to do the trick.

43081j commented:
@fb55 @wooorm there's probably some way we can make this build work but it'll likely involve changing how we produce the CJS output (not as simple as changing some flags, rather introducing something like tsup)

assuming im not missing something and that is the case, what do you two think about a new major which is ESM-only?

now that `require(esm)` is in all but one of node LTS versions (it isn't in 18 but that leaves LTS in april). maybe let's just publish a new major and people can gradually move to it as 18.x goes away?

cardoso commented:
@43081j for what it's worth, I'm super in favor of a new esm-only major version.

wooorm commented:
:+1: to esm only

cardoso commented:
Technically the error we get for build:cjs (TS7016) is not emit-related, so I reverted the changes to build:cjs script and added [--noCheck](https://www.typescriptlang.org/tsconfig/#noCheck).

@fb55 are you able to confirm whether this change is non-breaking for Cheerio?

fb55 commented:
> @fb55 are you able to confirm whether this change is non-breaking for Cheerio?

Yup, non-breaking for Cheerio.

fb55 commented:
Thanks @cardoso!

43081j commented:
@danshome this is because `entities` relies on node module resolution in 6.x, since it ships a dual package

if you set your tsconfig's `moduleResolution` to `node16`, it should work

the types do exist, its just that the mode your typescript is running in doesn't read `exports` in the `package.json`

@fb55 up to you if you want people to have to do this, or update `entities` to be ESM only and export real paths (i.e. set `decode.js` to point at `decode.js`, the actual on-disk path).
